File: code-examples/discountRule.js
function calculateDiscount(orderTotal, isLoyaltyMember) {
    if (orderTotal > 100 && isLoyaltyMember) {
        return orderTotal * 0.10;
    }
    return 0.0;
}

// Example usage
const total = 120;
const loyalty = true;
const discount = calculateDiscount(total, loyalty);
console.log(`Discount: $${discount.toFixed(2)}`);

File: code-examples/discount_rule.py
def calculate_discount(order_total, is_loyalty_member):
    """Returns the discount amount for the order."""
    if order_total > 100 and is_loyalty_member:
        return order_total * 0.10
    return 0.0

# Example usage
if __name__ == "__main__":
    total = 120
    loyalty = True
    discount = calculate_discount(total, loyalty)
    print(f"Discount: ${discount:.2f}")
